src/unsplash.js

```diff
@@ -36,16 +38,23 @@ async function meta(src, clientid) {
   const qs = {
     client_id: clientid,
   };
-  return request({ uri: `https://api.unsplash.com/photos/${id}`, qs, json: true });
+  // const ret1 = await request({uri:`https://api.unsplash.com/photos/${id}`, qs, json: true });
+  return fetch(`https://api.unsplash.com/photos/${id}?client_id=${qs.client_id}`);
 }
 
 async function decorator(metadata, options) {
   const enriched = { ...metadata };
   const src = metadata.twitter_card.url;
+  const resp = await meta(src, options.UNSPLASH_AUTH);
+
+  if (!resp.ok) {
+    return new Error(`Statuscode: ${resp.status} with status test: ${resp.statusText}`);
+  }
 
+  /* eslint-disable camelcase */
   const {
     user, urls, alt_description, width,
-  } = await meta(src, options.UNSPLASH_AUTH);
+  } = await resp.json();
```

Comment:
why don't you use `meta()` as before?

Reply:
True, it does make more sense for all of the meta stuff to happen there!

Reply:
@trieloff @tripodsan correct me if my thinking is off here, but I guess I wanted to try to avoid awaiting the request immediately; i was hoping maybe allowing for the call stack to pop off the current call; and return to the calling function, would mean more time for the request to return so this would mean less awaiting?

Reply:
versus waiting for the request; then after getting the result; returning to the calling function?